### PR TITLE
provider/aws: adds the ability to apply tags to spot request instances

### LINKED ISSUE #3263: aws_spot_instance_request tags won't apply to instance
I have an `aws_spot_instance_request` like this:

``` hcl
resource "aws_spot_instance_request" "seleniumgrid" {
    ami = "${var.amiPuppet}"
    key_name = "${var.key}"
    instance_type = "c3.4xlarge"
    subnet_id = "${var.subnet}"
    vpc_security_group_ids = [ "${var.securityGroup}" ]
    user_data = "${template_file.userdata.rendered}"
    wait_for_fulfillment = true
    spot_price = "${var.price}"
    availability_zone = "${var.zone}"
    instance_initiated_shutdown_behavior = "terminate"
    root_block_device {
      volume_size = 100
      volume_type = "gp2"
    }
    tags {
      Name = "${var.name}.${var.domain}"
      Provider = "Terraform"
      CA_TEAM = "${var.team}"
      CA_ROLE = "${var.role}"
      CA_SERVICE = "${var.service}"
    }
}
```

The tags are being applied only to the spot request itself, not to the underlying instance. Is this an expected behavior? How can I change this?

Thanks!

### PR DESCRIPTION
**Pull request by richardbowden**
Fixes #3263

adds the ability to tag instances that are created by spot instance requests

### CODE REVIEW

File: builtin/providers/aws/tags.go [hunk 1]
```diff
@@ -58,6 +64,33 @@ func setElbV2Tags(conn *elbv2.ELBV2, d *schema.ResourceData) error {
 // setTags is a helper to set the tags for a resource. It expects the
 // tags field to be named "tags"
 func setTags(conn *ec2.EC2, d *schema.ResourceData) error {
+	var err error
+
+	id := d.Id()
+
+	// Check for a Spot Instance Requests as it requires special
+	// handeling, the tagging process will tag the spot request it self
```
Comment (kwilczynski) [inline builtin/providers/aws/tags.go]: Typo.

File: builtin/providers/aws/tags.go [hunk 2]
```diff
@@ -2,13 +2,19 @@ package aws
...
 	"log"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/elbv2"
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
+const (
```
Comment (kwilczynski) [inline builtin/providers/aws/tags.go]: I think, that we do not need a constant that is being used only once.

**Comment by kwilczynski** [comment]
@richardbowden thank you!

Would you be able to add acceptance tests? So that we can ensure that the tags are being set and things hold water?

**Comment by richardbowden** [comment]
@kwilczynski will make the changes mentioned :), will also get some tests done

**Comment by richardbowden** [comment]
tagging code paths are tested in two ways, the original path will be tested as before using the existing resource types that test tags, for example `TestAccAWSInstance_tags` which is located in `/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_instance_test.go`. Tagging for spot instance requests and the actual instance created are tested in `github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_spot_instance_request_test.go`

**Comment by richardbowden** [comment]
@stack72 hey, just checking to see if anything else needs to be done for this PR ? Happy to do more

**Comment by davidfic** [comment]
Any update on this PR? Has this been solved by another PR? This has been in conflict mode for a year.

**Comment by richardbowden** [comment]
I guess it needs to be redone due to it not being merged back when the PR was created and mergeable. does not seem to of been a feature that was needed.

**Comment by apparentlymart** [comment]
Hello @richardbowden, and thanks for working on this!

As part of the the Terraform 0.10 release earlier this year, all of the Terraform providers were moved to their own repositories in [the terraform-providers GitHub organization](https://github.com/terraform-providers), and removed from the Terraform Core repository.

Unfortunately due to the fact that new issues and pull requests are being opened constantly, it was not possible for the various provider maintainers to merge all outstanding pull requests before this split, and there is no automatic way to migrate a pull request to a new repository.

As a result, this pull request can sadly no longer be applied as-is, and so I'm going to close it.

If you or someone else has the time and motivation to apply same changes to [the `aws` provider repository](https://github.com/terraform-providers/terraform-provider-aws) and open a new PR there, the maintainers of that provider should be able to review and merge it.

Thanks again for working on this, and sorry it was not able to be merged before the provider repository changes.

**Pull request closed without merging**